### PR TITLE
DEV: Adds RSpec tests

### PR DESCRIPTION
**Pull request by Ahmedgagan**
@oblakeerickson

### CODE REVIEW

File: spec/components/guardian_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  let(:admin) { Fabricate(:user, trust_level: TrustLevel[0], admin: true) }
+  let(:moderator) { Fabricate(:user, trust_level: TrustLevel[0], moderator: true) }
+  let(:guardian) { Guardian.new(user) }
+  let(:guardian1) { Guardian.new(user1) }
+  let(:adminGuardian) { Guardian.new(admin) }
+  let(:moderatorGuardian) { Guardian.new(moderator) }
+
+  describe "can_upload_video?" do
+    it "returns true for users having specified trust level" do
+      expect(guardian.can_upload_video?).to eq(true)
```
Comment (Ahmedgagan) [inline spec/components/guardian_spec.rb:17]: hey @oblakeerickson, can you take a look at this? Here the method we added using `add_to_class`(can_upload_video?) is not called and it directly returns `nil` I tried very hard to find a solution, but can't get the reason why this is happening

Comment (oblakeerickson) [reply spec/components/guardian_spec.rb:17]: Yep, I'll take a look.

File: spec/components/guardian_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+describe Guardian do
+  let(:user) { Fabricate(:user, trust_level: TrustLevel[1]) }
+  let(:user1) { Fabricate(:user, trust_level: TrustLevel[0]) }
+  let(:admin) { Fabricate(:user, trust_level: TrustLevel[0], admin: true) }
+  let(:moderator) { Fabricate(:user, trust_level: TrustLevel[0], moderator: true) }
+  let(:guardian) { Guardian.new(user) }
+  let(:guardian1) { Guardian.new(user1) }
+  let(:adminGuardian) { Guardian.new(admin) }
+  let(:moderatorGuardian) { Guardian.new(moderator) }
+
```
Comment (oblakeerickson) [inline spec/components/guardian_spec.rb:14]:
```suggestion
before do
  SiteSetting.discourse_video_enabled = true
end
```

**Review by oblakeerickson — CHANGES_REQUESTED** [review]
I added a code suggestion. The plugin needs to be enabled ;) Don't worry it took me a bit to realize this too.

File: spec/components/guardian_spec.rb (hunk 2 above)
Comment (Ahmedgagan) [reply spec/components/guardian_spec.rb:14]: Thanks a lot 😊 that was very silly one 🤦‍♂️ I wasted a day on this

**Comment by Ahmedgagan** [comment]
@oblakeerickson Added a commit with the suggested change & some more RSpec tests.

**Review by oblakeerickson — APPROVED** [review]

**Pull request merged**